src/test/java/com/VetApp/PurrgentCare/services/PetServiceImplementationTest.java

```diff
@@ -72,4 +75,48 @@ public void addPet_whenValidInput_returnsValidInput() {
         // then
         verify(mockPetRepository, times(1)).save(fakePet);
     }
+
+
+    @Test
+    public void getAllPets_withValidInput_returnsAllPets() {
+        // given
+        final var countOfPets = new Random().nextInt(1000);
+        final var expected = buildPetList(countOfPets);
+        given(mockPetRepository.findAll())
+                .willReturn(expected);
+
+        // when
+        final var actual = serviceUnderTest.getAllPets();
+
+        // then
+        then(actual).isEqualTo(expected);
+        then(actual).hasSize(countOfPets);
+    }
+
+    @Test
+    public void getAllPets_whenNoPets_returnsEmptyList() {
+        // given
+        final List<Pet> expected = buildPetList(0);
+        given(mockPetRepository.findAll())
+                .willReturn(expected);
+
+        // when
+        final var actual = serviceUnderTest.getAllPets();
+
+        // then
+        then(actual).isEqualTo(expected);
+    }
+
+private List <Pet> buildPetList (Integer countOfPets) {
```

Comment:
Could leverage generics to pass in the object type with Pet/Person.

Future improvement, not blocking.